splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt

```diff
@@ -13,16 +17,31 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>() {
     override val viewModelClass: KClass<SplashViewModel>
         get() = SplashViewModel::class
 
-    override fun onViewModelCreated(viewModel: SplashViewModel) {
-        super.onViewModelCreated(viewModel)
-        val handler = Handler()
+    override fun onResume() {
+        super.onResume()
+        viewModel.percentage
+                .map { "Percentage : ${it * 100}" }
+                .bindToMain(percentText.text())
+                .disposedBy(disposeBag)
+        viewModel.navigateTo
+                .subscribe {
+            when (it) {
+                is Navigate.Forward -> {
+                    when (it.endpoint) {
+                        is SplashViewModel.NavigationEndpoint.Welcome -> {
+                            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
+                        }
+                    }
+                }
+                is Navigate.Back -> {
 
-        /**
-         * TODO:: Remove the post delay once splash is functional.
-         */
-        handler.postDelayed({
-            val intent = Intent(this, MainActivity::class.java)
-            startActivity(intent)
-        }, 1000)
+                }
+            }
+        }.disposedBy(disposeBag)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        disposeBag.clear()
```

Comment:
not sure we want to clear this out on pause. then the whole `disposebag` is shot when you come back to this screen. remove for now and we can discuss later.

Reply:
Can't really do that. since we are (for now) subscribing in  onResume() I need to clear out the dispose bag so that upon going to background, UI updates do not happen until we bring the app to foreground again.